src/test/run-fail/borrowck-local-borrow.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+// error-pattern:panic 1
+
+// revisions: ast mir
+//[mir]compile-flags: -Z emit-end-regions -Z borrowck-mir
+
+fn main() {
+    let x = 2;
+    let y = &x;
+    panic!("panic 1");
+}
```

Comment:
Can you also add the test that @arielb1 added to the issue and check that it generates two errors (one for each function) -- both in AST and MIR mode? [A quick test with play *seems* to suggest that it does not generate two errors today](https://play.rust-lang.org/?gist=9fab9b5b24c24187f27b72b6f380aa55&version=nightly), but it'd be good to verify that your test is not working before your patch, and works after while you are at it.

```rust
fn cplusplus_mode(x: isize) -> &'static isize { &x }

fn cplusplus_mode_exceptionally_unsafe(x: &mut Option<&'static mut isize>) {
    let z = 0;
    *x = Some(&mut z);
    panic!("catch me for a dangling pointer!")
}
```